pkg/controller/node/nodecontroller.go

```diff
@@ -351,6 +358,10 @@ func NewNodeController(
 func (nc *NodeController) Run() {
 	// Incorporate the results of node status pushed from kubelet to master.
 	go wait.Until(func() {
+		if !nc.nodeInformer.Informer().HasSynced() || !nc.podInformer.Informer().HasSynced() || !nc.daemonSetInformer.Informer().HasSynced() {
```

Comment:
That was my second though - I can do what you suggest, but then I'll need to do NC.Run() in a separate goroutine - which is semantically slightly bigger change than this one. As this is going to be cherry-picked I wanted to do safest one possible.